yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/reader/ReaderState.scala

```diff
@@ -2,9 +2,11 @@ package org.virtuslab.yaml.internal.load.reader
 
 sealed trait ReaderState
 case object ReaderState:
-  case object Stream                     extends ReaderState
-  case object Document                   extends ReaderState
-  final case class Mapping(indent: Int)  extends ReaderState
-  final case class Sequence(indent: Int) extends ReaderState
-  case object FlowMapping                extends ReaderState
-  case object FlowSequence               extends ReaderState
+  case object Stream   extends ReaderState
+  case object Document extends ReaderState
```

Comment:
`Stream` is unused and and `Document` has only one usage. If we get rid off them, `ReaderStateWithIndent` won't be necessary